lib/components/position_component.dart

```diff
@@ -99,6 +95,45 @@ abstract class PositionComponent extends Component {
     topLeftPosition = rect.topLeft.toVector2();
   }
 
+  @override
+  bool checkOverlap(Vector2 point) {
+    final corners = _rotatedCorners();
+    corners.add(corners.first);
+    for (int i = 1; i < corners.length; i++) {
+      final lastCorner = corners[i - 1];
```

Comment:
Nit, I would call this `previousCorner` first time I read this, I thought this was the last corner of the list and got me a little confused.

But totally optional comment.